api/src/main/java/com/epam/esm/config/SpringConfig.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.transaction.TransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+import org.springframework.validation.beanvalidation.MessageSourceResourceBundleLocator;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import javax.sql.DataSource;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+@Configuration
+@ComponentScan("com.epam.esm")
+@EnableWebMvc
+@PropertySource({
+        "classpath:${spring.profiles.active}.db.properties",
+        "classpath:application.properties"})
+@EnableTransactionManagement
+public class SpringConfig implements WebMvcConfigurer {
+    private final ApplicationContext applicationContext;
+
+    @Value("${application.gifts.datasource.jdbcUrl}")
+    private String jdbcUrl;
+
+    @Value("${application.gifts.datasource.driverClassName}")
+    private String driverClassName;
+
+    @Value("${application.gifts.datasource.name}")
+    private String username;
+
+    @Value("${application.gifts.datasource.password}")
+    private String password;
+
+    @Value("${application.gifts.datasource.minimumIdle}")
+    private Integer minimumIdle;
+
+    @Value("${application.gifts.datasource.maximumPoolSize}")
+    private Integer maximumPoolSize;
+
+    @Value("${message-source.basename}")
+    private String messageSourceBaseName;
+
+    @Autowired
+    public SpringConfig(ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    @Bean
+    public ObjectMapper objectMapper() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        return objectMapper;
+    }
+
+    @Bean
+    public DataSource dataSource() {
+        HikariConfig config = new HikariConfig();
+        config.setDriverClassName(driverClassName);
+        config.setJdbcUrl(jdbcUrl);
+        config.setUsername(username);
+        config.setPassword(password);
+        config.setMinimumIdle(minimumIdle);
+        config.setMaximumPoolSize(maximumPoolSize);
+        return new HikariDataSource(config);
+    }
+
+    @Bean
+    public JdbcTemplate jdbcTemplate() {
+        return new JdbcTemplate(dataSource());
```

Comment:
As you are executing method `dataSource()` will Spring create new DataSource? If yes, do we still need the DataSource created above?